app/models/provider.rb

```diff
@@ -19,13 +19,23 @@ def multiple_offices?
 
   class << self
     def from_omniauth(auth, office_codes)
-      find_or_initialize_by(auth_provider: auth.provider, uid: auth.uid).tap do |record|
-        record.update(
+      user = find_by(email: auth.info.email) ||
+             find_by(auth_provider: auth.provider, uid: auth.uid) ||
+             new
+
+      Rails.logger.info("Potential manual mapping needed for #{auth.info.email} (#{auth.uid})") if user.new_record?
+
+      user.tap do |record|
+        record.assign_attributes(
           email: auth.info.email,
           description: auth.info.description,
           roles: auth.info.roles,
           office_codes: office_codes,
+          auth_provider: auth.provider,
+          uid: auth.uid
```

Comment:
Not sure what the data looks like coming out of Entra ID e.g. roles and accounts/office codes - so can't assess whether or not this is correct (even though I know it probably is)

Could we document what the response looks like somewhere like confluence?

Reply:
As per a previous thread, the data comes from Omniauth and is identical to the SAML response with some extra fields.